src/commissaire_http/handlers/clusters.py

```diff
@@ -25,11 +29,222 @@ def list_clusters(message, bus):
+                message['params']['name']))
+    except Exception as error:
+        LOGGER.debug('Brand new cluster being created.')
+
+        if message['params'].get('network'):
+            # Verify the networks existence
+            try:
+                bus.request('storage.get', params=[
+                    'Network', {'name': message['params']['network']}
+                ])
+            except Exception as error:
+                # Default if the network doesn't exist
+                message['params']['network'] = C.DEFAULT_CLUSTER_NETWORK_JSON['name']  # noqa
+
+    try:
+        cluster = models.Cluster.new(**message['params'])
+        cluster._validate()
+        response = bus.request(
+            'storage.save', params=[
+                'Cluster', cluster.to_dict()])
+        return create_response(message['id'], response['result'])
+    except models.ValidationError as error:
+        return return_error(message, error, JSONRPC_ERRORS['INVALID_REQUEST'])
+
+
+def list_cluster_members(message, bus):
+    """
+    Lists hosts in a cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    response = None
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+        LOGGER.debug('Cluster found: {}'.format(cluster['result']['name']))
+        LOGGER.debug('Returning: {}'.format(response))
+        return create_response(
+            message['id'], result=cluster['result']['hostset'])
+    except Exception as error:
+        return return_error(message, error, JSONRPC_ERRORS['NOT_FOUND'])
+
+
+def update_cluster_members(message, bus):
+    """
+    Updates the list of members in a cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        old_hosts = set(message['params']['old'])  # Ensures no duplicates
+        new_hosts = set(message['params']['new'])  # Ensures no duplicates
+        LOGGER.debug('old_hosts="{}", new_hosts="{}"'.format(
+            old_hosts, new_hosts))
+    except Exception as error:
+        return return_error(message, error, JSONRPC_ERRORS['BAD_REQUEST'])
+
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+    except Exception as error:
+        return return_error(message, error, JSONRPC_ERRORS['NOT_FOUND'])
+
+    if old_hosts != set(cluster['result']['hostset']):
+        msg = 'Conflict setting hosts for cluster {0}'.format(
+            cluster['result']['name'])
+        LOGGER.error(msg)
+        return return_error(message, msg, JSONRPC_ERRORS['CONFLICT'])
+
```

Comment:
The `FIXME` comments in the old `on_put` code are still relevant, I think.  Would you mind pasting them here so they don't get forgotten?